cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -14,7 +15,46 @@ export class AwsDefaults {
         return first<AWSKeyPair>(keyPairs);
     };
 
-    static setDefaultNodeType = (nodeTypeList: string[], nodeProfile: string) => {
+    static setDefaultNodeType = async (nodeProfile: string) => {
+        const nodeTypeList: string[] = await AwsService.getAwsNodeTypes();
+        return AwsDefaults.defaultNodeTypeStategy(nodeTypeList, nodeProfile);
+    };
+
+    static setDefaultVpc = async (nodeProfile: string) => {
+        const result: { [key: string]: string }[] = [];
+        const azList: AWSAvailabilityZone[] = await AwsService.getAwsAvailabilityZones();
+        switch (nodeProfile) {
+            case AWS_NODE_PROFILE_NAMES.SINGLE_NODE:
+                {
+                    const defaultAZ: { [key: string]: string } = {};
+                    if (azList[0]['name'] !== undefined) {
+                        defaultAZ['name'] = azList[0]['name'];
+                        const azNodeTypeList = await AwsService.getAwsNodeTypes(azList[0]['name']);
+                        defaultAZ['workerNodeType'] = AwsDefaults.defaultNodeTypeStategy(azNodeTypeList, nodeProfile);
+                        defaultAZ['publicSubnetID'] = '';
+                        defaultAZ['privateSubnetID'] = '';
+                        result.push(defaultAZ);
+                    }
+                }
+                break;
+            default: {
+                [...azList].slice(0, 2).forEach(async (az: AWSAvailabilityZone) => {
+                    const defaultAZ: { [key: string]: string } = {};
+                    if (az['name'] !== undefined) {
+                        defaultAZ['name'] = az['name'];
+                        const azNodeTypeList = await AwsService.getAwsNodeTypes(az['name']);
+                        defaultAZ['workerNodeType'] = AwsDefaults.defaultNodeTypeStategy(azNodeTypeList, nodeProfile);
+                        defaultAZ['publicSubnetID'] = '';
+                        defaultAZ['privateSubnetID'] = '';
+                        result.push(defaultAZ);
+                    }
+                });
```

Comment:
I think it might be helpful to log an error to the console here if there is an AZ which doesn't have name defined. It shouldn't happen, but since we are checking to see if the name is undefined, it might be good to know that it happened

Reply:
Yes sir. About the error handler issue. Maybe we can create a new task and process all errors in the same way?